src/lando/ui/legacy/revisions.py

```diff
@@ -146,6 +173,64 @@ def post(self, request: WSGIRequest, revision_id: int) -> HttpResponse:
         return redirect(request.META.get("HTTP_REFERER"))
 
 
+class UpliftAssessmentLinkView(LandoView):
+    """Link an existing uplift assessment to a revision."""
+
+    @force_auth_refresh
+    @method_decorator(require_phabricator_api_key(optional=False, provide_client=False))
+    def post(self, request: WSGIRequest, revision_id: int) -> HttpResponse:
+        """Link an existing uplift assessment to this revision."""
+
+        uplift_revision = UpliftRevision.one_or_none(revision_id=revision_id)
+        existing_assessment = uplift_revision.assessment if uplift_revision else None
+
+        link_form = LinkUpliftAssessmentForm(request.POST, user=request.user)
+
+        if not link_form.is_valid():
+            errors = [
+                f"{field}: {', '.join(field_errors)}"
+                for field, field_errors in link_form.errors.items()
+            ]
+
+            for error in errors:
+                messages.add_message(request, messages.ERROR, error)
+
+            return redirect(request.META.get("HTTP_REFERER"))
+
+        assessment = link_form.cleaned_data["assessment"]
+        previous_assessment = existing_assessment
+
+        with transaction.atomic():
+            uplift_revision, created = UpliftRevision.objects.update_or_create(
+                revision_id=revision_id,
+                defaults={"assessment": assessment},
+            )
+
+        if previous_assessment and previous_assessment.pk == assessment.pk:
+            messages.add_message(
+                request,
+                messages.INFO,
+                "This revision is already linked to the selected assessment.",
+            )
+        else:
+            set_uplift_request_form_on_revision.apply_async(
```

Comment:
Could we do this inline rather than async? This should be relatively quick (and it otherwise increases the risk of a race like I mentioned above [slightly]).